backend/inspirehep/rt/tickets.py

```diff
@@ -178,18 +172,80 @@ def get_users():
     Returns:
         list (dict): list of all users as {id, name} dict
     """
-    queues = current_cache.get("rt_users")
-    if queues:
-        return queues
-    else:
-        queues = _get_all_of("user")
-        if queues:
-            current_cache.set(
-                "rt_users",
-                queues,
-                timeout=current_app.config.get("RT_USERS_CACHE_TIMEOUT", 86400),
-            )
-        return queues
+    return _get_from_cache("rt_users", partial(_get_all_of, "user"))
+
+
+def get_rt_user_by_email(email, force_update=False):
+    """Find user info for specified e-mail in RT.
+
+    Args:
+        email(str): Email of user to look for.
+        force_update(bool): Forces to query users from RT
+          even if they are available in cache.
+
+    Returns:
+        dict: dictionary of the user {
+            "id":"user/<id>",
+            "Name":"<name>",
+            "EmailAddress": "<e-mail>"
+        }
+
+    """
+
+    def _parse_users_with_emails(lines):
+        data = zip(lines[0::3], lines[1::3], lines[2::3])
+        users = []
+        for user_data in data:
+            if user_data and all(user_data):
+                users.append(
+                    dict(
+                        [map(str.strip, element.split(":", 1)) for element in user_data]
+                    )
+                )
+        return users
+
+    def _get_rt_users():
+        lines = query_rt("user", fields=["Name", "EmailAddress"])
+        _users_data = _parse_users_with_emails(lines)
+        if not _users_data:
+            raise EmptyResponseFromRT
+        return _users_data
+
+    users_data = _get_from_cache("rt_users_with_emails", _get_rt_users, force_update)
+    result = [user for user in users_data if user["EmailAddress"] == email]
+
+    if not result and not force_update:
+        return get_rt_user_by_email(email, force_update=True)
+    elif not result:
+        raise NoUsersFound
+    return result[0]
```

Comment:
Instead of that why we don't catch `NoUsersFound` in the caller function and call there `get_rt_user_by_email(email, force_update=True)`? Should be part of the business logic?

Reply:
Why you want to expose low level implementation of get_user (cache should work behind the scenes) to some high level methods? This will mean that if we would like to use it in other places, we should implement cache handling in every place where we use it (remember that this cache is empty when app starts for the first time so every update/restart will cause cache to be empty...)